### ISSUE TITLE
Pushing password on website still '"Pushing"

### ISSUE THREAD

**Issue by grulicht** [pglombardo/PasswordPusher#1193]
Hello,

I have a problem:

Pushing password on website still '"Pushing", but in log of docker container I see path for create password for example "emf3kuvvyebu" and when I use my domain and path password i push everybody, but not automatically like in your pwpush web after "Push it" password on website. I do not know where is a problem.

Thanks for tips.

**Comment by jnbalbona** [comment]
Hi, I have the same problem!
Ubuntu + Docker + Nginx is my architecture.

Inspecting the requests from Chrome, even though the PWP__OVERRIDE_BASE_URL variable is properly set (https://...), the system is returning a URL via http:, which is being blocked by the browser:

Mixed Content: The page at 'https://desa*****.****.com/' was loaded over HTTPS, but requested an insecure resource 'https://desa*****.****.com/es/p/hgkoics401s/preview'. This request has been blocked; the content must be served over HTTPS.

If you change the blocked URL to https, in another browser tab it displays correctly and the link provided there is correct.

Is there any way to solve this?
Thank you in advance.

**Comment by selkopb** [comment]
Hello, 

I have exactly the same problem with an Ubuntu + Docker + Apache (as reverse proxy).
Do you have a suggestion for a resolution?

**Comment by grulicht** [comment]
@selkopb I do not know solution and still I checking on solution. :)

**Comment by pglombardo** [comment]
Hi all – apologies for the late reply. I'm traveling for the next couple weeks, but I'll take a closer look once I get back.  Summer vacations...

**Comment by kayvanaarssen** [comment]
Having the same issue, retried many times. Did not dig into it deeper but seems logic with it returning HTTP.

Also having an issue with file uploads, think that's the same issue or related.

When I upload for exampe 3 files, it does the first or maybe 2. And then gets stuck so to say.

Tried in many browsers, and also incognito. Same issue.

Just wanted to start using Password Pusher. Hope that this will be fixed asap, so we can start using it.

**Comment by pglombardo** [comment]
Hi @kayvanaarssen are you also running nginx in front of your Password Pusher instance?

**Comment by kayvanaarssen** [comment]
Yes, with the Nginx Proxy Manager that comes with the predefined Docker Configs.
Something we need to change in there then?

**Comment by pglombardo** [comment]
I'm not sure what the issue is yet but I there are a few issues (e.g. #794) that are all running with nginx in the front end.  I suspect there is a configuration issue that is causing all of these issues.  Needs investigation though.

Can you reproduce the issue when accessing the backend app directly?

**Comment by kayvanaarssen** [comment]
I'm on holiday now, and deleted the server that it was running on so cannot test it now🙈
But cannot believe that others are not running it with NGINX.
Is this something you can test in the meantime. We have used your templates for docker and but in HTTPS in the .evn
I'm back on 08-08-2023 so can test again at that point.

**Comment by pglombardo** [comment]
For sure it works without nginx as pwpush.com doesn't use nginx and the local test suite either.  I've tested with nginx in the past but not recently.

No rush - enjoy your vacation.  I'll loop back to this hopefully soon.

**Comment by kayvanaarssen** [comment]
@pglombardo Just tested without https/nginx just ip:5100 same issue...

![image](https://github.com/pglombardo/PasswordPusher/assets/672841/528dcf26-4beb-47ec-bb23-0cecaa22ad70)

**Comment by pglombardo** [comment]
Thanks @kayvanaarssen.  Really odd.  pwpush.com works, my local container tests work, the built in test suite that mocks a browser works...

If I can eventually reproduce this, I'm sure it'll be easy to fix.  A few questions:

1.  What container/tag are you using?
2. Are there any messages in the browser javascript console?
3. What browser?
4. Can you attach the container logs?  You can set `PWP__LOG_TO_STDOUT=true` to get all the request logs.

Much appreciated

**Comment by kayvanaarssen** [comment]
See here also Browser console. This is with local storage.
Also we are using S3 but not amazon that should work right?

1. Now 1.29.0 (going to try latest now)
2. See screenshot
3. Brave but chrome as well
4. See below

![image](https://github.com/pglombardo/PasswordPusher/assets/672841/98aa24a9-72a7-44ec-818e-75b10c6686c1)


```
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] warden (1.2.9) lib/warden/manager.rb:36:in `block in call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] warden (1.2.9) lib/warden/manager.rb:34:in `catch'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] warden (1.2.9) lib/warden/manager.rb:34:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/tempfile_reaper.rb:15:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/etag.rb:27:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/conditional_get.rb:40:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/head.rb:12:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/http/permissions_policy.rb:38:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/http/content_security_policy.rb:36:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/session/abstract/id.rb:266:in `context'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/session/abstract/id.rb:260:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/cookies.rb:704:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] activesupport (7.0.6) lib/active_support/callbacks.rb:99:in `run_callbacks'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rollbar (3.4.0) lib/rollbar/middleware/rails/rollbar.rb:25:in `block in call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rollbar (3.4.0) lib/rollbar.rb:145:in `scoped'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rollbar (3.4.0) lib/rollbar/middleware/rails/rollbar.rb:22:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/debug_exceptions.rb:28:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rollbar (3.4.0) lib/rollbar/middleware/rails/show_exceptions.rb:22:in `call_with_rollbar'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/show_exceptions.rb:29:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] lograge (0.12.0) lib/lograge/rails_ext/rack/logger.rb:18:in `call_app'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] railties (7.0.6) lib/rails/rack/logger.rb:25:in `block in call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] activesupport (7.0.6) lib/active_support/tagged_logging.rb:99:in `block in tagged'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] activesupport (7.0.6) lib/active_support/tagged_logging.rb:37:in `tagged'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] activesupport (7.0.6) lib/active_support/tagged_logging.rb:99:in `tagged'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] railties (7.0.6) lib/rails/rack/logger.rb:25:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/remote_ip.rb:93:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] request_store (1.5.1) lib/request_store/middleware.rb:19:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack-timeout (0.6.3) lib/rack/timeout/core.rb:148:in `block in call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack-timeout (0.6.3) lib/rack/timeout/support/timeout.rb:19:in `timeout'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack-timeout (0.6.3) lib/rack/timeout/core.rb:147:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/request_id.rb:26:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/method_override.rb:24:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/runtime.rb:22:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/executor.rb:14:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/static.rb:23:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack (2.2.7) lib/rack/sendfile.rb:110:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] actionpack (7.0.6) lib/action_dispatch/middleware/host_authorization.rb:131:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] rack-cors (2.0.1) lib/rack/cors.rb:102:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] railties (7.0.6) lib/rails/engine.rb:530:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] railties (7.0.6) lib/rails/railtie.rb:226:in `public_send'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] railties (7.0.6) lib/rails/railtie.rb:226:in `method_missing'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] puma (6.3.0) lib/puma/configuration.rb:270:in `call'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] puma (6.3.0) lib/puma/request.rb:100:in `block in handle_request'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] puma (6.3.0) lib/puma/thread_pool.rb:344:in `with_force_shutdown'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] puma (6.3.0) lib/puma/request.rb:99:in `handle_request'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] puma (6.3.0) lib/puma/server.rb:443:in `process_client'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] puma (6.3.0) lib/puma/server.rb:245:in `block in run'
pwpush-pwpush-1  | [d7feaa7f-8f37-4cac-8c18-2027bd1a49e9] puma (6.3.0) lib/puma/thread_pool.rb:151:in `block in spawn_thread'
pwpush-db-1      | 2023-08-09 14:28:34+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.0.2+maria~ubu2204 started.
pwpush-db-1      | 2023-08-09 14:28:35+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
pwpush-db-1      | 2023-08-09 14:28:35+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.0.2+maria~ubu2204 started.
pwpush-db-1      | 2023-08-09 14:28:35+00:00 [Note] [Entrypoint]: MariaDB upgrade not required
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] Starting MariaDB 11.0.2-MariaDB-1:11.0.2+maria~ubu2204 source revision 0005f2f06c8e1aea4915887decad67885108a929 as process 1
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Using transactional memory
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Number of transaction pools: 1
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] mariadbd: O_TMPFILE is not supported on /tmp (disabling future attempts)
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Completed initialization of buffer pool
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Opened 3 undo tablespaces
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: 128 rollback segments in 3 undo tablespaces are active.
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: log sequence number 245492; transaction id 538
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] Plugin 'FEEDBACK' is disabled.
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] Plugin 'wsrep-provider' is disabled.
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] Server socket created on IP: '0.0.0.0'.
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] Server socket created on IP: '::'.
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] InnoDB: Buffer pool(s) load completed at 230809 14:28:35
pwpush-db-1      | 2023-08-09 14:28:35 0 [Note] mariadbd: ready for connections.
pwpush-db-1      | Version: '11.0.2-MariaDB-1:11.0.2+maria~ubu2204'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
```

**Comment by pglombardo** [comment]
> Errno::EACCES (Permission denied @ dir_s_mkdir - /opt/PasswordPusher/storage/7n)

At least for the local storage, there was a permissions issue.  When uploading the file, it couldn't be written.  The specific error is with a `mkdir`.  Was that a mounted volume or really inside the container?

**Comment by kayvanaarssen** [comment]
Does not matter really.
1. Local and no volume mounted = same issue
2. S3 Storage (S3 Compatiple) for example Wasabi (using an other provider in this case) but should all be the same.
3. With mounted volume I let docker create the locations did not do anything with the rights

It works for mysql it created the folders in there without an issue.

Any idea what could cause this?

**Comment by pglombardo** [comment]
Ok I think I have a few ideas:

I was able to reproduce this with local storage.  I ran `chmod 555` and `find . -type d -exec chmod 555 {} \;` on/in the `/storage` directory and I got the same red upload bar. 

<img width="1058" alt="Screenshot 2023-08-09 at 17 08 22" src="https://github.com/pglombardo/PasswordPusher/assets/395132/ba4533e8-8de7-479e-9add-14bf5f2dc959">

So whenever you see the red upload bar and a greyed out "Pushing..." button, it means that it couldn't write the file (for whatever reason).

I think what we're missing is a CORS configuration for the S3 buckets.  For reference, [this is the related documentation](https://edgeguides.rubyonrails.org/active_storage_overview.html#cross-origin-resource-sharing-cors-configuration) for the upload library.

Could you try and add this CORS configuration to your S3 buckets?

```
[
  {
    "AllowedHeaders": [
      "Content-Type",
      "Content-MD5",
      "Content-Disposition"
    ],
    "AllowedMethods": [
      "PUT"
    ],
    "AllowedOrigins": [
      "https://www.example.com" <<< replace with your URL
    ],
    "MaxAgeSeconds": 3600
  }
]
```

**Comment by kayvanaarssen** [comment]
Strange right that this happens even in the local storage option while Password Pusher (docker) creates the folders itself...

As for S3 all sources are allowed. I can also access it via Transmit on mac or cyberduck.
Also other applications like veaam etc. And backup tools are using the same S3 service other buckets ofcourse but the service is working.
So can it be something else? As for region we do not have to provide that with the S3 provider that we use so we do not specify. But we can, an everything is accepted also tried that no luck there.

They use Ceph for S3 storage and make it public.

**Comment by pglombardo** [comment]
> Strange right that this happens even in the local storage option while Password Pusher (docker) creates the folders itself...

Yeah this is an odd issue that is hard to nail down.  I get the sense it's a mix of issues so my best idea is to fix each smaller issue as I find them.

Just to confirm, did you try the CORS configuration?  I don't doubt it works with the other tools.  I think CORS is specifically when an upload is started from your browser by being referred from a third party URL.

**Comment by kayvanaarssen** [comment]
Where do we need to add that exactly.
We do not have control over the S3 part we just get a an account where we can create a bucket. No config possible so to say.
Do we need to set it in Password Pusher somewhere?

**Comment by pglombardo** [comment]
In the S3 provider dashboard.  If that's not possible, let me see if I can figure another option.

**Comment by kayvanaarssen** [comment]
Right its not possible there. And like I wrote. Other thinks work without an issue. Also for example Nextcloud to point out an other more usual product. The service is from: dadup.eu

**Comment by kayvanaarssen** [comment]
Okay,,, just tested.

```
chmod 777 pwpusherfiles
```
That's the folder on the host, that I mapped to the Volume.
But... Here comes the fun part.
I've not specified `local` in the .env but `amazon`

```bash
PWP__FILES_STORAGE="amazon"
PWP__FILES__S3__ENDPOINT="nl.dadup.eu"
PWP__FILES__S3__ACCESS_KEY_ID=""
PWP__FILES__S3__SECRET_ACCESS_KEY=""
PWP__FILES__S3__REGION='us-east-1'
PWP__FILES__S3__BUCKET="pwpush-files-1"
```

**Comment by pglombardo** [comment]
That's odd - are you sure you restarted the application?

Thanks for the S3 creds.  I just substituted out the credentials for yours and I got a CORS error:

<img width="864" alt="Screenshot 2023-08-09 at 18 46 23" src="https://github.com/pglombardo/PasswordPusher/assets/395132/c6522ae3-59da-467b-93bc-7b1cd5f64bea">

**Comment by kayvanaarssen** [comment]
Yes, I do a restart of the Docker after I change anything.
Alright, so than I need to ask the provider if they can set it. Never had an issue with this before on their side.
Is there something we can do on the Password Pusher side to address this issue?
Also i'm testing mine from a public Domain, and also have set this in the .env file.

**Comment by pglombardo** [comment]
I thought about removing the direct upload feature although it would be non-trivial.

Direct uploads really help when people are uploading multi gigabyte files.  Without it, it would be equivalent of the application undergoing a denial of service - 1 dedicated process to handle a 5 minute upload.  5-10 people do that and the application would be unresponsive.

Hopefully you can set that in your bucket.  I've updated the Configuration document to explain this requirement.

I'm AFK for the next couple hours but will be back later.  Thanks for all the help @kayvanaarssen!

**Comment by kayvanaarssen** [comment]
Even tried it with Digital Ocean Spaces. And set CORS.
No luck either.

![image](https://github.com/pglombardo/PasswordPusher/assets/672841/bd18f991-54f5-448f-b9e3-3e994800e410)

**Comment by kayvanaarssen** [comment]
Got it working now...

```
PWP__FILES__STORAGE='amazon'
PWP__FILES__S3__ENDPOINT='https://ams3.digitaloceanspaces.com'
PWP__FILES__S3__ACCESS_KEY_ID='ACCESS_KEY_HERE'
PWP__FILES__S3__SECRET_ACCESS_KEY='SECRET_ACCESS_KEY_HERE'
PWP__FILES__S3__BUCKET='BUCKET_NAME_HERE'
```
Needed to set the ENDPOINT with `https://`

**Comment by kayvanaarssen** [comment]
Tested some more with BackBlaze B2 as well. There we set the CORS Header. That's not enough.

![image](https://github.com/pglombardo/PasswordPusher/assets/672841/ae90b9d7-f4b9-488b-a5d9-60497c5df951)
![image](https://github.com/pglombardo/PasswordPusher/assets/672841/eaa9532b-88a6-4f30-b4f7-458578b7cc93)

**Comment by grulicht** [comment]
And how is solution my problem on first message in this topic:

Hello,

I have a problem:

Pushing password on website still '"Pushing", but in log of docker container I see path for create password for example "emf3kuvvyebu" and when I use my domain and path password i push everybody, but not automatically like in your pwpush web after "Push it" password on website. I do not know where is a problem.

Thanks for tips.

**Comment by kayvanaarssen** [comment]
Are you running the latest version?
And which version with or without DB?

**Comment by selkopb** [comment]
Hello, 

Yes i'm using the last stable version, I just pulled the new version and the problem is still the same.
I'm using the version without database. 

Here is my docker-compose file : 
[docker-compose.txt](https://github.com/pglombardo/PasswordPusher/files/12394535/docker-compose.txt)

And the Apache config used as reverse proxy 
[apache config.txt](https://github.com/pglombardo/PasswordPusher/files/12394527/apache.config.txt)

But it's working when I use http

**Comment by jnbalbona** [comment]
Same as @selkopb but with nginx.

Nginx config
    server {
        listen 443 ssl;
        server_name pwp.xxxxxx.xx;
        if ($host != 'pwp.xxxxxx.xx'){
           return 403;
        }
        ssl_certificate /etc/certs/cert.pem;
        ssl_certificate_key /etc/certs/privkey.key;

        location / {
            allow  all;
            
            charset utf-8;
            client_max_body_size 50M;
		
            proxy_set_header Host $http_host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_pass http://192.168.47.100:5100/;
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
        }
		
        error_page 403 /error/403.html;
        error_page 404 /error/404.html;
        error_page 500 502 503 504 /error/50x.html;
    }

**Comment by WarpRat** [comment]
@jnbalbona - did you ever fix your issue? I'm having the same problem using local storage. We just attempted to update from 1.14 which was working fine but now see the same http/https mismatch when attempting to push. I tried setting `PWP__HOST_PROTOCOL` even though it's supposed to default to `https` but that didn't seem to help. I've also attempted setting both `PWP__OVERRIDE_BASE_URL` and `FORCE_SSL`. Neither of those seemed to help either.

**Comment by pglombardo** [comment]
Related update for "Stuck on 'Pushing...'" in #794:

> This morning I tested the [Docker & Nginx example](https://github.com/pglombardo/PasswordPusher/tree/master/containers/examples/pwpush-and-nginx) setup from scratch and I got the same stuck on "Pushing.." error (File uploads with pwpush-ephemeral and local storage).  After a bit of digging, I found that File uploads were getting blocked by per second throttling.
> 
> If you are having trouble uploading files, please try setting `PWP__THROTTLING__SECOND=20` and please post back your results.
> 
> With this, I was able to upload files working without issue.
> 
> When I switched to cloud storage (Amazon S3) I got the CORs error in the browser console but was fixed by applying the right CORs policy to the S3 buckets.  Documentation [here](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-file-pushes).
> 
> There is more but I don't want to overwhelm the thread.  I put other findings in the [Troubleshooting wiki page](https://github.com/pglombardo/PasswordPusher/wiki/Troubleshooting).
> 
> I hope that this helps at least some of you here.

**Comment by jnbalbona** [comment]
Hi.
Sorry for the lack of knowledge, could you give an example of using a functional NGINX configuration?
Thanks in advance.

**Comment by jnbalbona** [comment]
I answer myself:
    server {
        listen 443 ssl;
        server_name pwp.xxxxx.xx;
        if ($host != 'pwp.xxxxx.xx'){
           return 403;
        }
        ssl_certificate /etc/certs/cert.pem;
        ssl_certificate_key /etc/certs/privkey.key;

        location / {
            allow  all;
            
            charset utf-8;
            client_max_body_size 50M;
		
            proxy_set_header Host $http_host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_pass http://<IP>:<Puerto>/;
 	    **proxy_set_header X-Forwarded-Proto https;**
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
        }
		
        error_page 403 /error/403.html;
        error_page 404 /error/404.html;
        error_page 500 502 503 504 /error/50x.html;
    }

Thank you very much anyway.

**Comment by selkopb** [comment]
Hello, 
@jnbalbona, I was just about to post my feedback.
I finally managed to correct my problem by adding the following configuration to the Apache configuration: 
```
    RequestHeader set X-Forwarded-Host "pwpush.example.fr"
    RequestHeader set X-Forwarded-Port "443
    RequestHeader set X-Forwarded-Proto "https"
```

I also remove the SSL-related options in my docker-compose file: 
```
PWP__OVERRIDE_BASE_URL=https://pwpush.example.fr/
FORCE_SSL=true
```

Thank you very much for the help anyway !

**Comment by pglombardo** [comment]
> Sorry for the lack of knowledge, could you give an example of using a functional NGINX configuration?

For reference there is an example Password Pusher with nginx setup [here](https://github.com/pglombardo/PasswordPusher/tree/master/containers/examples/pwpush-and-nginx).

**Comment by WarpRat** [comment]
After more investigation we determined that our configuration of an `AWS NLB terminating SSL > nginx ingress controller > pwpush` attaches an `X-Forwarded-Proto: http` header. We're changing some of our ingress configuration to allow us to properly set that header for pwpush traffic but it would be great if there was a configuration option that would take precedence over the header for environments where it would be more difficult to make that change.

**Comment by pglombardo** [comment]
> it would be great if there was a configuration option that would take precedence over the header for environments where it would be more difficult to make that change.

The other option is `PWP__OVERRIDE_BASE_URL` which will [take precedence over the HTTP headers](https://github.com/pglombardo/PasswordPusher/blob/43e91040c1e0f33d087bb587fc1760b137e0d644/app/helpers/application_helper.rb#L23).  For reference, the documentation on this is [here](https://github.com/pglombardo/PasswordPusher/blob/43e91040c1e0f33d087bb587fc1760b137e0d644/app/helpers/application_helper.rb#L23).

This setting assures that generated secret URLs are generated with that base URL.  This doesn't apply to the HTTP back and forth between the application and the proxy.

**Comment by WarpRat** [comment]
Thanks for the response. We had already experimented with setting that override base url setting but the issue was the communication between the backend and proxy. The generate secret would fail because it was attempting to request http resources from an https page. I think we could have made it work with the `FORCE_SSL` option and some changes to how the pwpush server was running too but I see that option is marked as deprecated, hopefully before it gets removed there will be some option to available to override the protocol that gets used for the application <> proxy communication. Like I said, we fixed our issue with some network reconfiguration but I could see this being an issue for other folks down the road. Regardless, I appreciate the quick responses here and all your work on this tool, overall it's great and really helpful for keeping our secrets safe ❤️

**Comment by pglombardo** [comment]
I believe this issue should be resolved for most.  If so I'd like to close this issue.

But to summarise:

1. Stuck on "Pushing" was often caused by the built in throttling which was set too low.  The latest versions of the application now [have larger throttling thresholds](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#throttling).  Details in [this comment](https://github.com/pglombardo/PasswordPusher/issues/1193#issuecomment-1694256990).
2. @kayvanaarssen [with a great investigation](https://github.com/pglombardo/PasswordPusher/issues/1193#issuecomment-1672020896) identified potential issues with CORS configuration.  This issue may still exist.  If so, we should break it out into a new issue entirely so we can focus solely on that topic.  Please let me know if this issue still exists for you and if so, create a new issue solely for that topic.
3. Proxy <--> Password Pusher communication is difficult when it doesn't work.  Right now we offer [proxy configuration documentation](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#proxies) and a [section on force_ssl](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#forcing-ssl-links).  It's minimal but not ideal.

Thanks everyone for your patience, help and input.

Waiting for feedback, but I'll aim to close this issue soon if I don't hear anything.

**Comment by kayvanaarssen** [comment]
@pglombardo I'll try to setup a clone of our existing one and test like I did before and let you know.

**Comment by pglombardo** [comment]
@kayvanaarssen let's break that testing out to a separate issue specifically for CORS.  Could you file a new issue with your findings?  Even a copy/paste would be appreciated.

**Comment by cyril23** [comment]
> I finally managed to correct my problem by adding the following configuration to the Apache configuration:

thanks a lot! same thing helped me, with Ubuntu + Docker + Apache Reverse Proxy configuration via `RewriteRule`

```
	RewriteEngine On
	RewriteRule ^(.*) http://127.0.0.1:5100/$1 [P,L]
	# fix proxy headers: https://github.com/pglombardo/PasswordPusher/issues/1193#issuecomment-1695532432
	RequestHeader set X-Forwarded-Port "443"
	RequestHeader set X-Forwarded-Proto "https"
	RequestHeader set X-Forwarded-Host "mysubdomain.mywebsite.com"
```

before this, I thought I needed to edit the `location` header manually via `Header edit Location ^http://(.*)$ https://$1`, but actually the root of the problem were these missing HTTP Headers, see https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#forcing-ssl-links
> Note: This is a legacy setting and is no longer suggested for use. **If using a proxy, make sure to have your proxy forward the X-Forwarded-Host, X-Forwarded-Port and X-Forwarded-Proto HTTP headers**. [...]

**Comment by nov1c3** [comment]
In case someone uses F5 BIG-IP in front.

Create a custom HTTP profile to insert X-Forwarded-Proto:https header (as per K43444200) and attach it to the VS.  
No need to specify PWP__HOST_PROTOCOL and/or PWP__OVERRIDE_BASE_URL.

Thanks!

**Comment by pglombardo** [comment]
Thank you @nov1c3 for the help!

I believe this issue should be fully resolved now.  Is anyone here still experiencing this issue?

**Comment by bigbeka** [comment]
> Thank you @nov1c3 for the help!
> 
> I believe this issue should be fully resolved now. Is anyone here still experiencing this issue?

@pglombardo thanks, unless I `chmod 777` the storage folder, files are not uploading after ‘Push It!’ What permissions do I need to give? The storage is Bind mount through docker compose, the user that is running docker has write permission to the storage bind mount.

Any other troubleshooting I can be doing?

**Comment by pglombardo** [comment]
The pwpush container is a non-root container - that might have something to do with it.

Inside the container, the application [runs with](https://github.com/pglombardo/PasswordPusher/blob/master/containers/docker/Dockerfile#L87-L88) `uid:1000` & `gid:1000` and user `pwpusher`.  Open an [app console](https://docs.pwpush.com/docs/application-console/) and check ownership of the storage directory and try to touch a file in that directory.

That would probably be the best place to start.  It most likely has to do with something between the container identity and the external identity on the volume.

Hope this helps.

**Comment by Yudis53732** [comment]
Yes

Pada Sen, 2 Sep 2024 01.34, Peter Giacomo Lombardo ***@***.***>
menulis:

**Comment by bigbeka** [comment]
> The pwpush container is a non-root container - that might have something to do with it.
> 
> Inside the container, the application [runs with](https://github.com/pglombardo/PasswordPusher/blob/master/containers/docker/Dockerfile#L87-L88) `uid:1000` & `gid:1000` and user `pwpusher`. Open an [app console](https://docs.pwpush.com/docs/application-console/) and check ownership of the storage directory and try to touch a file in that directory.
> 
> That would probably be the best place to start. It most likely has to do with something between the container identity and the external identity on the volume.
> 
> Hope this helps.

Thanks for your rely. I will try.

**Comment by Yudis53732** [comment]
Thanks, I'll check it out.

Pada Sen, 2 Sep 2024 01.52, bigbeka ***@***.***> menulis:

**Comment by pglombardo** [comment]
Hi all - do any issues remain here?  I'll close out this issue eventually unless I hear otherwise.

To summarize a few key points:

Stuck "Pushing" button can happen [if a proxy doesn't send the X-Forwarded-* headers](https://docs.pwpush.com/docs/proxies/#required-proxy-headers) or if using file uploads, [CORS isn't properly setup](https://docs.pwpush.com/docs/self-hosted-configuration/#file-pushes).  There is some great investigation above.  Checking browser console for errors is key.

In older versions of Password Pusher, the rate limiter was to aggressive and would block file uploads in the back and forth requests to upload a file.  This has since been fixed.

If anyone is still having any issues, please feel free to open a new issue and we can tackle remaining individual issues separately.

Thanks all for the help & input!

**Comment by pglombardo** [comment]
One off topic update here but for any users that use F5 BIG-IP, CloudFlare or any external proxy:

A security research team reported a vulnerability related to proxies [here](https://github.com/pglombardo/PasswordPusher/security/advisories/GHSA-ffp2-8p2h-4m5j).

In [v1.49.0](https://github.com/pglombardo/PasswordPusher/releases/tag/v1.49.0) only local network proxies are trusted by default.  For that release and going forward, when using a proxy that is not on the local network, you will likely have to specifically authorize the external IP to act as a proxy.  Documentation [here](https://docs.pwpush.com/docs/proxies/#trusted-proxies).

If you have any questions, let me know.

**Comment by DediCATeD88** [comment]
> Hi all - do any issues remain here? I'll close out this issue eventually unless I hear otherwise.
> 
> To summarize a few key points:
> 
> Stuck "Pushing" button can happen [if a proxy doesn't send the X-Forwarded-* headers](https://docs.pwpush.com/docs/proxies/#required-proxy-headers) or if using file uploads, [CORS isn't properly setup](https://docs.pwpush.com/docs/self-hosted-configuration/#file-pushes). There is some great investigation above. Checking browser console for errors is key.
> 
> In older versions of Password Pusher, the rate limiter was to aggressive and would block file uploads in the back and forth requests to upload a file. This has since been fixed.
> 
> If anyone is still having any issues, please feel free to open a new issue and we can tackle remaining individual issues separately.
> 
> Thanks all for the help & input!


Hi,

regarding this issue using nginx reverse proxy:

We are unable to set:

proxy_set_header  X-Forwarded-Port $server_port;
proxy_set_header  X-Forwarded-Host $host;
proxy_set_header  X-Forwarded-Proto $scheme;
proxy_set_header  X-Forwarded-Ssl on;

We set:

      PWP__OVERRIDE_BASE_URL: 'https://**************.de'
      PWP__HOST_PROTOCOL: 'https'

Via docker compose.

Unfortunately still no luck:

Mixed Content: The page at 'https:// **************.de/' was loaded over HTTPS, but requested an insecure resource 'http:// **************.de/p/1**************eq/preview'. This request has been blocked; the content must be served over HTTPS.

Any ideas?

**Comment by pglombardo** [comment]
As a final summary and resolution for this issue:  See [this comment](https://github.com/pglombardo/PasswordPusher/issues/1193#issuecomment-2480498559).

I'll close this issue out but if anyone has further problems, please file a new issue and we'll pick up from there.  Thanks all for the input/help in investigating!
